js-rattler/crate/package_name.d.ts

```diff
@@ -0,0 +1,188 @@
+/** A unique symbol used for branding `PackageName` types. */
+declare const PACKAGE_NAME_BRAND: unique symbol;
+
+/**
+ * A **branded type** representing a validated package name.
+ *
+ * - This type is **enforced at runtime** using `isPackageName()`.
+ * - Ensures that a package name conforms to the expected format.
+ *
+ * @example
+ *
+ * ```ts
+ * const pkg: PackageName = "valid-package" as PackageName;
+ * ```
+ *
+ * @public
+ */
+export declare type PackageName = string & {
+    [PACKAGE_NAME_BRAND]: void;
+};
+
+/**
+ * Defines the allowed characters for any package name.
+ *
+ * Allowed characters:
+ *
+ * - Lowercase letters (`a-z`)
+ * - Uppercase letters (`A-Z`)
+ * - Digits (`0-9`)
+ * - Underscore (`_`)
+ * - Dash (`-`)
+ * - Dot (`.`)
+ *
+ * @public
+ */
+export declare type PackageNameChar =
```

Comment:
Would it be better to use a regex?

Reply:
Yeah, all the runtime functions use a regex, but the type definitions cannot use regexes so we have to write them out unfortunately.